src/main/java/teammates/logic/core/EmailGenerator.java

```diff
@@ -84,6 +85,27 @@
         return emails;
     }
     
+    /**
+     * Generates the feedback submission confirmation email for the given {@code session} for {@code student}
+     * and {@code instructor}.
+     */
+    public EmailWrapper generateFeedbackSubmissionConfirmationEmails(
+            FeedbackSessionAttributes session, StudentAttributes student,
+            InstructorAttributes instructor) {
+        CourseAttributes course = coursesLogic.getCourse(session.getCourseId());
+        EmailWrapper email = new EmailWrapper();
+        
+        if (instructor != null) {
+            email = generateFeedbackSessionEmailBaseForInstructorSubmissionConfirmation(course, session,
+                    instructor);
+        }
+        if (student != null) {
+            email = generateFeedbackSessionEmailBaseForStudentSubmissionConfirmation(course, session,
+                    student);
+        }
+        return email;
```

Comment:
There are some problems here:
- If both `student` and `instructor` are non-null then the email for the instructor is lost.
- If both `student` and `instructor` are null the `email` will be an empty wrapper.
See if it can be made such that one and only one of `student` and `instructor` are null.